### PR TITLE
fix: add missing properties to schema

### PR DESCRIPTION
**Pull request by pottekkat**
### Description

Adds missing properties to the schema so that it can be used for validating configuration files more accurately.

Also consolidates all the schemas to a single schema instead of having to validate with multiple schemas.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by pottekkat** [comment]
I'm unsure why the tests are failing.

File: apisix/cli/schema.lua [hunk 1]
```diff
@@ -246,18 +405,225 @@ local config_schema = {
...
+                            type = "string"
+                        },
+                        client_header_timeout = {
+                            type = "string"
+                        },
+                        client_body_timeout = {
+                            type = "string"
+                        },
+                        client_max_body_size = {
+                            type = "number"
```
Comment (Revolyssup) [inline apisix/cli/schema.lua:573]:
```suggestion
                            type = "string"
```

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: apisix/cli/schema.lua (hunk 1 above)
Comment (pottekkat) [reply apisix/cli/schema.lua:573]: Let me try. I originally had it as string but it threw an error.

[`f3ecf05` (#10339)](https://github.com/apache/apisix/pull/10339/commits/f3ecf055e025c1b92abc4a0804a9403e31ea8a68)

**Comment by pottekkat** [comment]
@Revolyssup I got this error before so I tried changing it:

https://github.com/apache/apisix/actions/runs/6549717930/job/17787244082?pr=10339#step:5:287

**Comment by pottekkat** [comment]
@Revolyssup Should we add a case where it can be either a string or a number?

**Comment by monkeyDluffy6017** [comment]
LGTM, That's a pretty big change. So let's make the ci pass

File: apisix/cli/schema.lua [hunk 2]
```diff
@@ -302,110 +683,87 @@ local config_schema = {
...
-            properties = {
-                items = {
-                    properties = {
-                        name = {type = "string"},
-                        key = {type = "string"},
-                        role = {type = "string"},
+            dependencies = {
+                role = {
+                    oneOf = {
+                        {
```
Comment (Revolyssup) [inline apisix/cli/schema.lua:709]: @pottekkat Will you make sure that this dependency schema fits right with this[1] yaml snippet here?
1. https://github.com/apache/apisix/blob/88406dceab26ed7f25b663088b0b5e760bd45a38/t/cli/test_validate_config.sh#L210C1-L210C1

**Review by Revolyssup — CHANGES_REQUESTED** [review]

**Comment by pottekkat** [comment]
I reverted to the same changes as in the last commit where the tests were passing in this commit: https://github.com/apache/apisix/pull/10339/commits/bf8bdb87649dc18cabd85624ce301d83991e65a2

But now, the tests fail.

I'm now trying by removing all changes and checking if the tests pass now and build from there.

**Comment by pottekkat** [comment]
All checks pass with these changes.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Comment by monkeyDluffy6017** [comment]
@Revolyssup please help to check

**Review by Revolyssup — APPROVED** [review]

**Pull request merged**